firmware/common2015/drivers/cc1201/CC1201Config.cpp

```diff
@@ -0,0 +1,1212 @@
+                    expected = config->modemStatus0;
+                    break;
+
+                case CC1201EXT_MARC_STATUS1:
+                    expected = config->marcStatus1;
+                    break;
+
+                case CC1201EXT_MARC_STATUS0:
+                    expected = config->marcStatus0;
+                    break;
+
+                case CC1201EXT_PA_IFAMP_TEST:
+                    expected = config->paIfampTest;
+                    break;
+
+                case CC1201EXT_FSRF_TEST:
+                    expected = config->fsrfTest;
+                    break;
+
+                case CC1201EXT_PRE_TEST:
+                    expected = config->preTest;
+                    break;
+
+                case CC1201EXT_PRE_OVR:
+                    expected = config->preOvr;
+                    break;
+
+                case CC1201EXT_ADC_TEST:
+                    expected = config->adcTest;
+                    break;
+
+                case CC1201EXT_DVC_TEST:
+                    expected = config->dvcTest;
+                    break;
+
+                case CC1201EXT_ATEST:
+                    expected = config->atest;
+                    break;
+
+                case CC1201EXT_ATEST_LVDS:
+                    expected = config->atestLvds;
+                    break;
+
+                case CC1201EXT_ATEST_MODE:
+                    expected = config->atestMode;
+                    break;
+
+                case CC1201EXT_XOSC_TEST1:
+                    expected = config->xoscTest1;
+                    break;
+
+                case CC1201EXT_XOSC_TEST0:
+                    expected = config->xoscTest0;
+                    break;
+
+                case CC1201EXT_AES:
+                    expected = config->aes;
+                    break;
+
+                case CC1201EXT_MDM_TEST:
+                    expected = config->mdmTest;
+                    break;
+
+                default:
+                    expected = 0x00;
+            }
+
+            // check for a configuration mismatch
+            if (device->readReg(extReg, EXT_FLAG_ON) != expected) {
+                // to_string doesn't seem to be supported by the mbed
+                char regCStr[2];
+                char valCStr[2];
+                char expCStr[2];
+                sprintf(regCStr, "%02X", extReg);
+                sprintf(valCStr, "%02X", device->readReg(extReg, EXT_FLAG_ON));
+                sprintf(expCStr, "%02X", expected);
+                string regStr((const char*)regCStr, 2);
+                string valStr((const char*)valCStr, 2);
+                string expStr((const char*)expCStr, 2);
+                string errorStr = ("ExtReg: " + regStr + ", Val: " + valStr +
+                                   ", Exp: " + expStr);
+                // LOG(OK, "%s", errorStr.c_str());
+                // push the error onto the queue for processing later
+                // configurationFaults.push(errorStr);
```

Comment:
All we really care about at this point is the part number register at startup. Could possibly read back the frequency regs to see if the configuration was written successfully, but that's about it now that there's pretty reliable reg read/write code.